### ISSUE TITLE
Multiple canonical relation type 

### ISSUE THREAD

**Issue by YoucTagh** [w3c/dx-connegp#44]
In both [Example 19](https://w3c.github.io/dx-connegp/connegp/#eg-qsa-keydiscovery) and [Example 20](https://w3c.github.io/dx-connegp/connegp/#eg-qsa-keydisco-altprofiles), there is a `Link` header with two `canonical` relation type. Is this something acceptable?

The guidelines in [RFC](https://datatracker.ietf.org/doc/html/rfc6596#section-3) that introduce this relation type suggest that it is not:

```
- Specify only one canonical link relation for a resource.
```

**Comment by YoucTagh** [comment]
This comment is a summary of the discussion @youctagh, @larsgsvensson and @rob-metalinkage  had on this issue during [the meeting on 09 August 2023](https://www.w3.org/2023/08/09-dxwg-minutes.html).

### Problem: 
In a response using the `Link` header, we need a way to express *the default representation* as well as *the alternative representations*. In addition, we need to express the default or preferred representation within a dimension or combination of dimensions, not just media type and profile (as mentioned in [Section 7.3.2.2](https://w3c.github.io/dx-connegp/connegp/#qsa-tokens) 
> ... other content negotiation dimensions (language etc.)

also in [Section 7.3.4](https://w3c.github.io/dx-connegp/connegp/#qsa-keydiscovery) 

>... indicate Media Type and other content negotiation dimensions.

### Example:
Suppose we have a resource **A** identified by [http://example.org/resource/a](http://example.org/resource/a) that has six representations (r<sub>i</sub>) that vary along three dimensions: media type (m), language (l), and profile (p):

* The first representation r<sub>1</sub> has media type m<sub>1</sub>, language l<sub>1</sub> and conforms to profile<sub>1</sub> (identified by [http://example.org/resource/a?m=m1&p=p1&l=l1](http://example.org/resource/a?m=m1&p=p1&l=l1)).
* r<sub>2</sub>: m<sub>1</sub>, l<sub>3</sub>, p<sub>2</sub> (identified by [http://example.org/resource/a?m=m1&p=p2&l=l3](http://example.org/resource/a?m=m1&p=p2&l=l3))
* r<sub>3</sub>: m<sub>2</sub>, l<sub>2</sub>, p<sub>1</sub> (identified by [http://example.org/resource/a?m=m2&p=p1&l=l2](http://example.org/resource/a?m=m2&p=p1&l=l2))
* r<sub>4</sub>: m<sub>3</sub>, l<sub>2</sub> (profile is unknown) (identified by [http://example.org/resource/a?m=m3&l=l2](http://example.org/resource/a?m=m3&l=l2))
* r<sub>5</sub>: m<sub>3</sub> (language and profile are unknown) (identified by [http://example.org/resource/a?m=m3](http://example.org/resource/a?m=m3))
* r<sub>6</sub>: m<sub>1</sub>, l<sub>2</sub>, p<sub>1</sub> (identified by[http://example.org/resource/a?m=m1&p=p1&l=l2](http://example.org/resource/a?m=m1&p=p1&l=l2))

We want to express that:
* r<sub>1</sub> is the global default representation (if no default representation is defined for a dimension or combination of dimensions).
* r<sub>1</sub> is the preferred representation for media type m<sub>1</sub> (i.e. if the request only specifies a preference for media type m<sub>1</sub>, then between r<sub>1</sub>, r<sub>2</sub> and r<sub>6</sub> that have this media type, r<sub>1</sub> is the preferred one).
* r<sub>3</sub> is the preferred representation for the profile p<sub>1</sub> (i.e. between r<sub>1</sub>, r<sub>3</sub> and r<sub>6</sub>, r<sub>3</sub> is the preferred one).
* r<sub>6</sub> is the preferred representation when the combination m<sub>1</sub> and p<sub>1</sub> is requested.
* r<sub>3</sub> is the preferred representation when the combination p<sub>1</sub> and l<sub>2</sub> is requested.

### Proposed solution: 
Use the relation types `canonical` and `alternate` (check [IANA Link Relations](https://www.iana.org/assignments/link-relations/link-relations.xhtml)). The former is used to define the global default representation, the latter to define an alternative representation that is neither a global default nor a dimension specific preferred representation (in our example, r<sub>2</sub> and r<sub>3</sub> and r<sub>4</sub>).
We introduce a new relation type `preferred` to define the preferred representation for a dimension or combination of dimensions, which is used with the new attribute `dim` to specify the dimension of variability. 

The Link headers to express the example are:

```
Link:
    <http://example.org/resource/a?m=m1&p=p1&l=l1>;
        rel="canonical";
        type="m1";        format="p1";        lang="l1",

    <http://example.org/resource/a?m=m1&p=p1&l=l1>;
        rel="preferred";
        dim="m";
        type="m1";        format="p1";        lang="l1",

    <http://example.org/resource/a?m=m1&p=p2&l=l3>;
        rel="alternate";
        type="m1";        format="p2";        lang="l3",

    <http://example.org/resource/a?m=m2&p=p1&l=l2>;
        rel="preferred";
        dim="p";
        type="m2";        format="p1";        lang="l2",

    <http://example.org/resource/a?m=m3&l=l2>;
        rel="alternate";
        type="m3";        lang="l2",

    <http://example.org/resource/a?m=m3>;
        rel="alternate";
        type="m3",

    <http://example.org/resource/a?m=m1&p=p1&l=l2>;
        rel="preferred";
        dim="m p";
        type="m1";        format="p1";        lang="l2",

    <http://example.org/resource/a?m=m2&p=p1&l=l2>;
        rel="preferred";
        dim="p l";
        type="m2";        format="p1";        lang="l2"


```

### Notes
Note that the solution to this problem may also affect other examples such as Examples [22](https://w3c.github.io/dx-connegp/connegp/#eg-altr-ext), [23](https://w3c.github.io/dx-connegp/connegp/#eg-altr) and [24](https://w3c.github.io/dx-connegp/connegp/#eg-altr-multi).

**Comment by rob-metalinkage** [comment]
To do:

remove multiple canonical tags from example.

add or refer to text to define precedence over dimensions.

No plans to implement a more nuanced "prefered" tag, so this will determine the degree of expressivity available.

**Comment by larsgsvensson** [comment]
I reopen and tag as "due-for-closing"

**Comment by rob-metalinkage** [comment]
Resolved in PR #46
